src/TreeSitter/Cursor.hs

```diff
@@ -0,0 +1,30 @@
+module TreeSitter.Cursor where
+
+import Data.Int
+import Data.Word
+import Foreign.C
+import Foreign.Ptr
+import TreeSitter.Node
+
+-- | A cursor for traversing a tree.
+--
+--   Note that we do not define 'Eq', 'Ord', or 'Storable' instances, as the underlying @TSTreeCursor@ type is not usefully copyable.
+data Cursor = Cursor
```

Comment:
My preferred idiom for this is `newtype Cursor = Cursor ()` to enforce that Cursor contains no useful Haskell-side information. But this is probably fine.

Reply:
We’ve done that elsewhere, but I’d kind of like to migrate away from it because it seems like an abuse of `newtype`s 😅